### PR TITLE
fix: missing quote in table name

### PR DESCRIPTION
**Pull request by averikitsch**
All table names should be wrapped in quotes to support special characters like dashes

### CODE REVIEW

File: src/langchain_google_cloud_sql_pg/vectorstore.py [hunk 1]
```diff
@@ -461,7 +461,7 @@ async def __query_collection(
         search_function = self.distance_strategy.search_function
 
         filter = f"WHERE {filter}" if filter else ""
-        stmt = f"SELECT *, {search_function}({self.embedding_column}, '{embedding}') as distance FROM {self.table_name} {filter} ORDER BY {self.embedding_column} {operator} '{embedding}' LIMIT {k};"
+        stmt = f"SELECT *, {search_function}({self.embedding_column}, '{embedding}') as distance FROM \"{self.table_name}\" {filter} ORDER BY {self.embedding_column} {operator} '{embedding}' LIMIT {k};"
```
Comment (jackwotherspoon) [inline src/langchain_google_cloud_sql_pg/vectorstore.py:464]: nit: for consistency should we follow the escape pattern of the other variables (embedding) in the query and use single quotes? or should they also be using backslashes?
```suggestion
        stmt = f"SELECT *, {search_function}({self.embedding_column}, '{embedding}') as distance FROM '{self.table_name}' {filter} ORDER BY {self.embedding_column} {operator} '{embedding}' LIMIT {k};"
```

File: src/langchain_google_cloud_sql_pg/vectorstore.py [hunk 2]
```diff
@@ -742,7 +742,7 @@ async def aapply_vector_index(
         params = "WITH " + index.index_options()
         function = index.distance_strategy.index_function
         name = name or index.name
-        stmt = f"CREATE INDEX {'CONCURRENTLY' if concurrently else ''} {name} ON {self.table_name} USING {index.index_type} ({self.embedding_column} {function}) {params} {filter};"
+        stmt = f"CREATE INDEX {'CONCURRENTLY' if concurrently else ''} {name} ON \"{self.table_name}\" USING {index.index_type} ({self.embedding_column} {function}) {params} {filter};"
```
Comment (jackwotherspoon) [inline src/langchain_google_cloud_sql_pg/vectorstore.py:745]: ditto here, would single quote be more readable?
```suggestion
        stmt = f"CREATE INDEX {'CONCURRENTLY' if concurrently else ''} {name} ON '{self.table_name}' USING {index.index_type} ({self.embedding_column} {function}) {params} {filter};"
```

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]
nit: @averikitsch do you mind updating PR description with a one or two line quick summary of this bug and the fix. Makes this open-source project more more readable and friendly to external contributors/users who want to understand things.

File: src/langchain_google_cloud_sql_pg/vectorstore.py (hunk 1 above)
Comment (averikitsch) [reply src/langchain_google_cloud_sql_pg/vectorstore.py:464]: The table name needs to use double quotes: `<class 'asyncpg.exceptions.PostgresSyntaxError'>: syntax error at or near "'test_tableb5955ff2_5a51_460a_b674_105101e6436f'"` but the embeddings need to be wrapped in a single quote.

Comment (jackwotherspoon) [reply src/langchain_google_cloud_sql_pg/vectorstore.py:464]: awesome, good to know

**Review by jackwotherspoon — APPROVED** [review]
LGTM ✅

**Pull request merged**